### PR TITLE
remove Spring Cloud AWS Autoconfigure region handling

### LINKED ISSUE #29: Fix links in Readme
Some links to the Spring docs seem to be broken.

### LINKED ISSUE #30: Fix region handling
The recent change that makes use of the `AmazonWebserviceClientFactoryBean` to build the `AWSKMSClient` is unfortunately not compatible with  [AWS' DefaultRegionProviderChain](http://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/java-dg-region-selection.html#automatically-determine-the-aws-region-from-the-environment), as it always picks the default region `us-east-2`, if not configured using [spring-cloud-aws-autoconfigure](http://cloud.spring.io/spring-cloud-aws/spring-cloud-aws.html#_configuring_region). 
As it does not make sense to force people to use this extra library, it should always use the AWS DefaultRegionProviderChain and offer an optional configuration property to allow overwriting it.

### PR DESCRIPTION
**Pull request by harti2006**
Fixes #30 and changes README accordingly, which fixes #29

### CODE REVIEW

**Comment by harti2006** [comment]
:+1:

File: encryption-cli/README.md [hunk 1]
```diff
@@ -3,13 +3,19 @@ Installation
...
+Configure AWS [credentials](http://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-default)
+and [region](http://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/java-dg-region-selection.html#automatically-determine-the-aws-region-from-the-environment).
+
 Usage
 -----
 
 ### Encrypt
 
-    $ ./run.sh encrypt 'Hello World!' 9d9fca31-54c5-4df5-ba4f-127dfb9a5031
+    $ ./run.sh --encrypt-plaintext='Hello World!' --aws.kms.keyId='9d9fca31-54c5-4df5-ba4f-127dfb9a5031'
```
Comment (maschleg) [inline encryption-cli/README.md]: this arg is ``--encrypt.plaintext``. To be consistent in wording with ``decrypt.cipherText`` I would rename it to ``encrypt.plainText``

Comment (harti2006) [reply encryption-cli/README.md]: oops

**Comment by maschleg** [comment]
👍

**Comment by harti2006** [comment]
:+1:

**Pull request merged**